blockchain/blockdao/blob_store.go

```diff
@@ -22,4 +38,209 @@ type (
 		GetBlobsByHeight(uint64) ([]*types.BlobTxSidecar, []string, error)
 		PutBlock(*block.Block) error
 	}
+
+	// storage for past N-day's blobs, structured as blow:
+	// 1. Block height is used as key to store blobs. When a new blob is stored,
+	//    the expired time (current - 18 days) is updated and all blobs earlier
+	//    than this time will be deleted.
+	// 2. Two mappings are kept in a separate index: (1) key to blob's height
+	//    and tx hashes, and (2) tx hash to the blob's height. Here's an example:
+	//    304 --> { height: 311344
+	//              []hash: eddf9e61fb9d8f5111840daef55e5fde
+	//                      95977fb340867fa13f787df864ed5388 }
+	//    and:
+	//    eddf9e61fb9d8f5111840daef55e5fde --> {height: 311344}
+	//    95977fb340867fa13f787df864ed5388 --> {height: 311344}
+	//    The 2nd mapping is used to support GetBlob(hash) function. It converts
+	//    the requested blob hash to the height, and retrieves the blob.
+	//    When a new blob overwrites an old one, it will check the 1st mapping,
+	//    and delete the hashes of old blob correspondingly.
+	// 3. The maximum number of blobs is 6 for each block, so maximum data size
+	//    stored by a key is 131kB x 6 = 786kB. The maximum total size of the
+	//    entire blob storage is 786kB x 311040 = 245GB.
+	//
+	blobStore struct {
+		kvStore                         db.KVStore
+		totalBlocks                     uint64
+		currWriteBlock, currExpireBlock uint64
```

Comment:
`currExpireBlock` is confusing.

Reply:
`currExpireBlock = currWriteBlock - 18 days`
it keeps track of the earlier block height in the storage, and is used to determine the range of blobs to be deleted. For example, say `currExpireBlock = 2000`, then after 30 mins, we should delete blobs from 2000 to 2360, and after that, update `currExpireBlock = 2360`